mobile/src/Student/screens/StudentName/index.js

```diff
@@ -12,13 +12,18 @@ const StudentName = ({ navigation, gameSession, setTeamInfo }) => {
     const [lastName, setLastName] = useState("")
     const firstNameTextRef = useRef(null)
     const lastNameTextRef = useRef(null)
+    const [showErrorText, setShowErrorText] = useState(false)
 
     onNameSubmit = () => {
+        if (!firstName || !lastName) {
+            setShowErrorText(true)
+            return
```

Comment:
So essentially, the reason why I put the first if statement without the focus is that the focus of a null field causes the crash. Specifically, because firstNameTextRef/lastNameTextRef is null when the second and third if statements are entered, the call to either firstNameTextRef.focus()/lastNameTextRef.focus() gives the error 
"TypeError: firstNameTextRef.focus us not a function (In 'firstNameTextRef,focus()', 'firstNameTextRef.focus' is undefined)" I can look into why the focus is causing the crash to enable focus but this seems to fix the crashing for the moment

Reply:
Yeah, which is weird, not sure how the code gets here with `ref` to be `null`. But for preventing that, we can do something like:

```javascript
if (!firstName) {
    setShowErrorText(true)
    if  (firstNameTextRef) {
        firstNameTextRef.focus()
    }
    return
}
```

Same is true for the lastName.  Repeating the code might not feel alright, in which case you can also extract the logic into a helper method like:

```javascript
const validateIsNotEmpty = (text, textInputRef) => {
    if (text) {
        setShowErrorText(false)
        return true
    }

    if (textInputRef) {
        textInputRef.focus()
    }

    setShowErrorText(true)

    return false
}

```

And instead of your code changes and the existing one, you can combine them into one `if` as follow:

```javascript
if (!validateIsNotEmpty(firstName, firstNameTextRef) ||
    !validateIsNotEmpty(lastName, lastNameTextRef)) {
        return
    }
```

Reply:
`
const validateIsNotEmpty = (text, textInputRef) => {
        if (text) {
            setShowErrorText(false)
            return true
        }

        if (textInputRef) {
            textInputRef.focus()
        }

        setShowErrorText(true)

        return false
    }

    onNameSubmit = () => {
        if (!validateIsNotEmpty(firstName, firstNameTextRef) ||
            !validateIsNotEmpty(lastName, lastNameTextRef)) {
            return
        } ...
`
Strange, I tried the above but still throws error only now the error is "TypeError: textInputRef.focus is not a function (In 'textInputRef.focus()', 'textInputRef.focus' is undefined)"

Reply:
What's the full changes? You can also push the changes and happy to look into it

Reply:
Also, fyi, you can use three \` and then the name of the language you're using in markup and get syntax highlighting. Like:

\`\`\`javascript
// Your code snippet here
\`\`\`

Sometimes you can also omit the language name and the renderer is smart enough to guess the languange

Reply:
Thank you for the help, I just pushed!

Reply:
Is the code you pushed the one that's having the issue with the error message above or is that fixed now?

Reply:
It is the one with the error message

Reply:
Perfect, thanks! I tested it and pushed the fix in your branch.  You can see the [this commit](https://github.com/rightoneducation/righton-app/pull/512/commits/b385611a89f14e46eb9ef459245009be81b3bedc). Feel free to check it out and let me know if you have any questions. I tested it with both and only one first / last name being empty and now focus works as expected and never crashes